oak-benchmarks-elastic/src/main/java/org/apache/jackrabbit/oak/benchmark/ElasticFacetSearchTest.java

```diff
@@ -0,0 +1,82 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.benchmark;
+
+
+import org.apache.commons.io.FileUtils;
+import org.apache.jackrabbit.oak.Oak;
+import org.apache.jackrabbit.oak.benchmark.util.TestHelper;
+import org.apache.jackrabbit.oak.fixture.JcrCreator;
+import org.apache.jackrabbit.oak.fixture.OakRepositoryFixture;
+import org.apache.jackrabbit.oak.fixture.RepositoryFixture;
+import org.apache.jackrabbit.oak.jcr.Jcr;
+import org.apache.jackrabbit.oak.plugins.index.elastic.ElasticConnection;
+import org.apache.jackrabbit.oak.plugins.index.elastic.ElasticIndexDefinition;
+import org.apache.jackrabbit.oak.plugins.index.elastic.index.ElasticIndexEditorProvider;
+import org.apache.jackrabbit.oak.plugins.index.elastic.query.ElasticIndexProvider;
+import org.apache.jackrabbit.oak.plugins.index.nodetype.NodeTypeIndexProvider;
+import org.apache.jackrabbit.oak.plugins.index.property.PropertyIndexEditorProvider;
+import org.apache.jackrabbit.oak.plugins.index.search.ExtractedTextCache;
+
+import javax.jcr.Repository;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class ElasticFacetSearchTest extends FacetSearchTest {
+
+    private ElasticConnection coordinate;
+    protected String indexName;
+
+    ElasticFacetSearchTest(Boolean storageEnabled, ElasticConnection coordinate) {
+        super(storageEnabled);
+        this.coordinate = coordinate;
+    }
+
+    @Override
+    protected Repository[] createRepository(RepositoryFixture fixture) throws Exception {
+        indexName = TestHelper.getUniqueIndexName("elasticFacetTest");
+        Map propMap = new LinkedHashMap<String, Boolean>();
+        propMap.put(SEARCH_PROP, false);
+        propMap.put(FACET_PROP_1, true);
+        propMap.put(FACET_PROP_2, true);
+        if (fixture instanceof OakRepositoryFixture) {
+            return ((OakRepositoryFixture) fixture).setUpCluster(1, new JcrCreator() {
+                @Override
+                public Jcr customize(Oak oak) {
+                    ElasticIndexEditorProvider editorProvider = new ElasticIndexEditorProvider(coordinate,
+                            new ExtractedTextCache(10 * FileUtils.ONE_MB, 100));
+                    ElasticIndexProvider indexProvider = new ElasticIndexProvider(coordinate);
+                    oak.with(editorProvider)
+                            .with(indexProvider)
+                            .with(new PropertyIndexEditorProvider())
+                            .with(new NodeTypeIndexProvider())
+                            .with(new FacetSearchTest.FacetIndexInitializer(indexName, propMap,
+                                    ElasticIndexDefinition.TYPE_ELASTICSEARCH, getFaceMode()));
```

Comment:
this does not compile. It should be:
```suggestion
                                    ElasticIndexDefinition.TYPE_ELASTICSEARCH, getFacetMode()));
```